packages/core/src/service/plugin/index.js

```diff
@@ -162,6 +176,23 @@ export default class PluginService {
     ...this.plugins.layout.map(({ name }: Plugin) => name)
   ]
 
+  migratePluginState = (state: any, plugin: Plugin, dataVersion: string): ?Object => {
+    if (!plugin || !dataVersion || !semver.valid(dataVersion)) {
+      return state
+    }
+    const usedMigrations = plugin.migrations ? plugin.migrations.filter(m => semver.gt(m.version, dataVersion)) : []
+    usedMigrations.sort((a, b) => semver.lt(a.version, b.version) ? 1 : -1)
+    try {
```

Comment:
A particular reason for try/catch here?

Reply:
The migrations can be coming from 3RD party plugin plugin yet we are executing them in the core package. Might be a bit paranoid but as started as backend dev, I have this thing where I wrap 3rd party code in try catch. People might be installing lot of plugins from weird sources and it potentially makes sense for it not to throw down the whole editor. Maybe we could introduce ErrorBoundaries as well. Or maybe I am just ready for my tinfoil hat :D

Reply:
The idea is ok, but the execution needs improvement. We should definitely catch this in an error boundary. I think just throwing out a `console.error` and returning `undefined` (which is a very difficult type do handle) is not the correct approach here. Let's not catch the error and try to (in a different PR) introduce e.g. ErrorBoundary.

Reply:
Ok